ducktape/tests/session_context.py

```diff
@@ -83,4 +134,75 @@ def generate_results_dir(session_id):
     :type session_id: str
     :rtype: str
     """
-    return session_id + "-test-results"
\ No newline at end of file
+    return session_id + "-test-results"
+
+
+def mkdir_p(path):
+    """mkdir -p functionality.
+    :type path: str
+    """
+    try:
+        os.makedirs(path)
+    except OSError as exc: # Python >2.5
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+        else:
+            raise
+
+
+class TestContext(Logger):
```

Comment:
It's a bit odd for this to be in session_context.py. If you want to group things together to avoid the proliferation of files, I'd group TestContext into test.py and rename this file session.py (which makes it available for any other small session-related classes that we might end up creating).